src/app/view/components/dot-add-persona-dialog/dot-add-persona-dialog.component.spec.ts

```diff
@@ -0,0 +1,165 @@
+        'dot.common.dialog.accept': 'Accept',
+        'dot.common.dialog.reject': 'Cancel'
+    });
+
+    beforeEach(() => {
+        const siteServiceMock = new SiteServiceMock();
+
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                DotAddPersonaDialogComponent,
+                DotCreatePersonaFormComponent,
+                TestFieldValidationMessageComponent,
+                MockSiteSelectorComponent
+            ],
+            imports: [BrowserAnimationsModule, DotDialogModule, FileUploadModule],
+            providers: [
+                DotWorkflowActionsFireService,
+                DotHttpErrorManagerService,
+                { provide: DotMessageService, useValue: messageServiceMock },
+                { provide: LoginService, useClass: LoginServiceMock },
+                { provide: SiteService, useValue: siteServiceMock }
+            ]
+        });
+
+        fixture = TestBed.createComponent(DotAddPersonaDialogComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+        dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
+    });
+
+    it('should not be visible by default', () => {
+        expect(dotDialog).toBeNull();
+    });
+
+    describe('visible Dialog', () => {
+        beforeEach(() => {
+            component.visible = true;
+            fixture.detectChanges();
+            dotDialog = fixture.debugElement.query(By.css('dot-dialog'));
+        });
+
+        it('should set dialog attributes correctly', () => {
+            expect(dotDialog.componentInstance.header).toEqual('Add Persona');
+            expect(dotDialog.componentInstance.appendToBody).toBe(true);
+            expect(dotDialog.componentInstance.actions).toEqual({
+                accept: {
+                    label: 'Accept',
+                    disabled: true,
+                    action: jasmine.any(Function)
+                },
+                cancel: {
+                    label: 'Cancel',
+                    action: jasmine.any(Function)
+                }
+            });
+        });
+
+        it('should handle disable state of the accept button when form value change', () => {
+            spyOn(component, 'handlerFormValidState');
+            component.personaForm.isValid.emit(true);
+
+            expect(component.handlerFormValidState).toHaveBeenCalledWith(true);
+        });
+
+        it('should reset persona form, disable accept button and set visible to false on closeDialog', () => {
+            spyOn(component.personaForm, 'resetForm');
+            component.closeDialog();
+
+            expect(component.personaForm.resetForm).toHaveBeenCalled();
+            expect(component.visible).toBe(false);
+            expect(component.dialogActions.accept.disabled).toBe(true);
+        });
+
+        it('should call closeDialog on dotDialog hide', () => {
+            spyOn(component, 'closeDialog');
```

Comment:
Don't spy on component method, just assert what the called of that method does.

Reply:
in this case is ok to use it, since I just want to check that the attribute `hide` on `dot-dialog` have that function set.  The functionality of the `closeDialog ` is tested in the previous test.